test/net_http_test.rb

```diff
@@ -0,0 +1,241 @@
+      end
+    end
+  end
+
+  def test_get_type_1_is_protected
+    open_circuit!
+    assert_raises Net::CircuitOpenError do
+      Net::HTTP.get(URI('http://localhost:41235/200'))
+    end
+  end
+
+  def test_get_type_2_is_protected
+    open_circuit!
+    assert_raises Net::CircuitOpenError do
+      http = Net::HTTP.new("localhost", "41235")
+      http.get("/")
+    end
+  end
+
+  def test_get_type_3_is_protected
+    open_circuit!
+    assert_raises Net::CircuitOpenError do
+      uri = URI('http://localhost:41235/200')
+      Net::HTTP.get_response(uri)
+    end
+  end
+
+  def test_post_type_1_is_protected
+    open_circuit!
+    assert_raises Net::CircuitOpenError do
+      uri = URI('http://localhost:41235/200')
+      Net::HTTP.post_form(uri, 'q' => 'ruby', 'max' => '50')
+    end
+  end
+
+  def test_http_start_and_inner_methods_are_protected
+    open_circuit!
+
+    uri = URI('http://localhost:41235/200')
+    assert_raises Net::CircuitOpenError do
+      Net::HTTP.start(uri.host, uri.port) do |_|
+      end
+    end
+
+    close_circuit!
+    Net::HTTP.start(uri.host, uri.port) do |http|
+      open_circuit!
+      get_subclasses(Net::HTTPRequest).each do |action|
+        assert_raises(Net::CircuitOpenError, "#{action.name} did not raise a Net::CircuitOpenError") do
+          request = action.new uri
+          http.request(request)
+        end
+      end
+    end
+  end
+
+  def test_custom_raw_semian_options_work
+    yaml_sample_config = {}
+    yaml_sample_config["development"] = {}
+    yaml_sample_config["development"]["http_default"] =
+      {"tickets" => 1,
+       "success_threshold" => 1,
+       "error_threshold" => 3,
+       "error_timeout" => 10}
+    yaml_sample_config["development"]["http_localhost_41235"] =
+      {"tickets" => 1,
+       "success_threshold" => 1,
+       "error_threshold" => 3,
+       "error_timeout" => 10}
+    sample_env = "development"
+
+    semian_options_proc = proc do |semian_identifier|
+      if !yaml_sample_config[sample_env].key?(semian_identifier)
+        yaml_sample_config[sample_env]["http_default"]
+      else
+        yaml_sample_config[sample_env][semian_identifier]
+      end
+    end
+
+    with_semian_options(semian_options_proc) do
+      Net::HTTP.start("localhost", 41_235) do |http|
+        assert_equal yaml_sample_config["development"][http.semian_identifier], http.raw_semian_options
+      end
+      Net::HTTP.start("localhost", 41_234) do |http|
+        assert_equal yaml_sample_config["development"]["http_default"], http.raw_semian_options
+      end
+      assert_equal yaml_sample_config["development"]["http_default"], Semian::NetHTTP.raw_semian_options
+    end
+  end
+
+  def test_adding_custom_errors_work
+    with_custom_errors([::OpenSSL::SSL::SSLError]) do
+      http = Net::HTTP.new("localhost", 41_235)
+      assert http.resource_exceptions.include?(::OpenSSL::SSL::SSLError)
+    end
+  end
+
+  def with_semian_options(options)
```

Comment:
All of these methods should be `private` to make it clear where the tests stop and the helpers start.